code/clue.js

```diff
@@ -1,59 +1,353 @@
+  document.getElementsByTagName('img')[0].style.borderRadius = '50%'
+  const classKillerContent = document.getElementsByClassName('killer-content')[0];
+
+  //Changes the text color on the killer card, if the background color is white
+  if (mystery.killer.color === 'white') {
+    classKillerContent.getElementsByTagName('h3')[0].style.color = 'black';
+    const killerTagP = classKillerContent.getElementsByTagName('p');
+    let i = 0
+    for (i=0; i < killerTagP.length; i++) {
+      killerTagP[i].style.color = 'black';
+    }
+
+  } else {
+    //Changes back to white
+    classKillerContent.getElementsByTagName('h3')[0].style.color = 'white';
+    const killerTagP = classKillerContent.getElementsByTagName('p');
+    let i = 0
+    for (i=0; i < killerTagP.length; i++) {
+      killerTagP[i].style.color = 'white';
+    }
+  }
+}
+
+const generateWeapon = () => {
+
+  // Hides the loader and shows weapon title and content 
+  document.getElementById('weaponLoader').style.display = 'none';
+  document.getElementById('weaponTitle').style.display = 'block';
+  document.getElementById('weaponContent').style.display = 'flex';
+  
+  //Generates and shows a weapon, but only if the killer deck is picked first
+  if (mystery.killer === 'who') {
+    document.getElementById('weaponTitle').innerHTML = 'Please pick a killer first';
+  } else {
+    mystery.weapon = randomSelector(moreWeapons)
+    document.getElementById('weaponName').innerHTML = mystery.weapon.name
+    document.getElementById('weaponWeight').innerHTML = 'Weight: ' + mystery.weapon.weight
+    document.getElementById('weaponCard').style.background = '#575A53'
+    document.getElementById('weaponHeadline').style.display = 'none'
+    document.getElementById('weaponTitle').innerHTML = 'WEAPON'
+    document.getElementById('weaponDecoration').style.display = 'block'
+    document.getElementById('weaponDecoration').style.border = '0px'
+    document.getElementById('weaponContent').style.justifyContent = 'center'
+  }
+}
+
+const generateRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  // Hides the loader and shows killer title and content 
+  document.getElementById('roomLoader').style.display = 'none';
+  document.getElementById('roomTitle').style.display = 'block';
+  document.getElementById('roomContent').style.display = 'flex';
+
+  document.getElementById('roomName').innerHTML = mystery.room
+  document.getElementById('roomCard').style.background = '#a8979f'
+  document.getElementById('roomHeadline').style.display = 'none'
+  document.getElementById('roomTitle').style.display = 'block'
+  document.getElementById('roomDecoration').style.display = 'block'
+  document.getElementById('roomDecoration').style.border = '0px'
+  document.getElementById('roomContent').style.justifyContent = 'center'
+}
+
+const pickKiller = () => {
+  // This function will be invoked when you click on the killer card.
+
+  showKillerLoader();
```

Comment:
Nice that you put all the styling in one function and later called the function here, much cleaner looking code.